tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -769,12 +885,12 @@ def rebuild_index(self):
             refs.clear()
         self.use_segments.clear()
         self.def_segments.clear()
+        self.scope_segments.clear()
         self.visit(asts)
         self.use_segments.sort()
         self.def_segments.sort()
```

Comment:
Probably good to sort scope segments here as well.

Reply:
Per our offline conversation, sorting the scope segments actually breaks things, because the sort/binary search only takes into account the "start". It needs to handle both "start" and "end". This issue has nothing to do with this PR though, and Daniel is looking into it